### ISSUE TITLE
Filter by typo in data browser

### ISSUE THREAD

**Issue by joeroe** [xronos-ch/xronos.rails#297]
Would be useful, but hard to implement using the current architecture of the data browser (#165).

**Comment by MartinHinz** [comment]
That is right, but in an agile perspective, could we now not make a filter based on the materialized view? With something like:

```
@data = DataView.where("typochronological_units::text LIKE ?", search_value)
```

**Comment by joeroe** [comment]
Perhaps, I can't remember exactly where I ran into trouble trying to do that before. 

It looks like the current materialised data view turns typos into attributes of c14s (as 'periods' and 'typochronological units'), which is backwards compatible with the v1 API but not strictly speaking how the rails data model is structured. typos and c14s are supposed to be independent descendents of sites (via sample -> context). Therefore, a filter for `typo = 'Neolithic'` should be interpreted as 'give me sites/contexts/samples with Neolithic typos' (and by extension c14s belonging to those sites, as long as they meet any other applied other filters).

In both the materialised data view and the old unmaterialised one, we cheated by pretending c14s are the basic unit for all queries (i.e. DataView is structured as a `SELECT ... FROM c14`), which simplifies things as long as we're filtering on c14 attributes, but causes problem when we move to other parts of the data model. In reality the unit should probably be something in between c14s/typos/dendros and samples (measurement in the v1 data model or what I've been calling a chrons). Then we provide with the data browser with a flat table to work in in long format, e.g.

```
-----------------------------------------------------
| sample_id | chron_type | chron_id | [joined cols] |
-----------------------------------------------------
| 1         | c14        | 1        | [...]         |
| 2         | typo       | 1        | [...]         |
-----------------------------------------------------
```

Which looking at it now is just what you'd have if you had a chron model with a polymorphic association to c14s, typos, etc. Maybe that could be the way forward? Bringing back measurements, basically, which IIRC is what the data browser originally was based on (Clemens was clearly more foresighted than I am!)
